packages/fluentui/react-northstar-prototypes/src/prototypes/MenuList/MenuItemCheckbox.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { MenuItem } from './MenuItem';
+import { useMenuListContext } from './menuListContext';
+
+export const MenuItemCheckbox = (props: any) => {
+  const {
+    checkedValues: { [props.name]: checkedItems = [] },
+    onCheckedValuesChange,
+  } = useMenuListContext();
+  const { value, name } = props;
+
+  const checked = checkedItems.indexOf(value) !== -1;
+  const checkItem = () => [...checkedItems, value];
+  const unCheckItem = () => {
+    const newCheckedItems = [...checkedItems];
+    const index = checkedItems.indexOf(value);
+    if (index !== -1) {
+      newCheckedItems.splice(index, 1);
+    }
+
+    return newCheckedItems;
+  };
+
+  const onClick = () =>
+    !checked ? onCheckedValuesChange(name, checkItem()) : onCheckedValuesChange(name, unCheckItem());
```

Comment:
```suggestion
  const toggleCheck = () => {
    const newCheckedItems = [...checkedItems];
    const index = checkedItems.indexOf(value);
    if (index !== -1) {
      newCheckedItems.splice(index, 1);
    } else {
      newCheckedItems.push(value)
    }
    return newCheckedItems
  }

  const onClick = () => onCheckedValuesChange(name, toggleCheck());
```

Could it be simplified like this?

Reply:
yup, I like it :-)